# HG changeset patch
# User John Bieling <john@thunderbird.net>
# Date 1709831696 -3600
# Node ID c15d29688650c61ea401ae34656c62d92b8e6e4b
# Parent  aa9ce79b79dcf45952f8c44245db3f355fe09ed3
Bug 1877774 - Move call for threadPane.restoreSortIndicator() into onCreatedView(). r=vineet

Under certain conditions gDBView is not setup and calls to
threadPane.restoreSortIndicator() silently fail in
folderPane._onSelect().

Differential Revision: https://phabricator.services.mozilla.com/D200186

diff --git a/mail/base/content/about3Pane.js b/mail/base/content/about3Pane.js
--- a/mail/base/content/about3Pane.js
+++ b/mail/base/content/about3Pane.js
@@ -2582,10 +2582,8 @@ var folderPane = {
 
       threadPane.updateListRole(
         !gViewWrapper?.showThreaded && !gViewWrapper?.showGroupedBySort
       );
-      threadPane.restoreSortIndicator();
-      threadPaneHeader.onFolderSelected();
     }
 
     this._updateStatusQuota();
 
diff --git a/mail/base/content/mailCommon.js b/mail/base/content/mailCommon.js
--- a/mail/base/content/mailCommon.js
+++ b/mail/base/content/mailCommon.js
@@ -982,8 +982,10 @@ var dbViewWrapperListener = {
       for (const col of ThreadPaneColumns.getCustomColumns()) {
         gViewWrapper.dbView.addColumnHandler(col.id, col.handler);
       }
       window.threadPane.setTreeView(gViewWrapper.dbView);
+      window.threadPane.restoreSortIndicator();
+      window.threadPaneHeader.onFolderSelected();
       window.threadPane.isFirstScroll = true;
       window.threadPane.scrollDetected = false;
       window.threadPane.scrollToLatestRowIfNoSelection();
     }
